wire-ios/Wire-iOS/Sources/Components/Settings/SettingsProperty.swift

```diff
@@ -155,20 +160,22 @@ final class SettingsUserDefaultsProperty: SettingsProperty {
 }
 
 typealias GetAction = (SettingsBlockProperty) -> SettingsPropertyValue
-typealias SetAction = (SettingsBlockProperty, SettingsPropertyValue) throws -> Void
+typealias SetAction = (SettingsBlockProperty, SettingsPropertyValue, @escaping (Result<Void, any Error>) -> Void) throws -> Void
 
 /// Genetic block property
 final class SettingsBlockProperty: SettingsProperty {
+    
     var enabled: Bool = true
 
     let propertyName: SettingsPropertyName
     func value() -> SettingsPropertyValue {
         return self.getAction(self)
     }
 
-    func set(newValue: SettingsPropertyValue) throws {
-        try setAction(self, newValue)
+    func set(newValue: SettingsPropertyValue, resultHandler: @escaping (Result<Void, any Error>) -> Void) throws {
+        try setAction(self, newValue, resultHandler)
         NotificationCenter.default.post(name: Notification.Name(rawValue: propertyName.changeNotificationName), object: self)
+        resultHandler(.success(()))
```

Comment:
Why do you call the handler here if you pass it in the set action?

Reply:
Fixed it!